src/object/srv_obj.c

```diff
@@ -1507,13 +1507,23 @@ obj_local_rw_internal(crt_rpc_t *rpc, struct obj_io_context *ioc,
 			goto post;
 		}
 
-		if (ioc->ioc_coc->sc_props.dcp_csum_enabled) {
+		if (ioc->ioc_coc->sc_props.dcp_csum_enabled &&
+		    daos_csummer_initialized(ioc->ioc_coc->sc_csummer)) {
+			struct daos_csummer csummer = {0};
+
+			rc = daos_csummer_copy_inplace(&csummer, ioc->ioc_coc->sc_csummer);
+
+			if (rc != 0) {
+				rc = -DER_NOMEM;
```

Comment:
Why always convert the failure to -DER_NOMEM?

Reply:
Thank you ... I meant to remove that. Updated.